tardis/simulation/base.py

```diff
@@ -410,3 +411,138 @@ def from_config(cls, config, **kwargs):
                    convergence_strategy=config.montecarlo.convergence_strategy,
                    nthreads=config.montecarlo.nthreads)
 
+    @classmethod
+    def from_hdf5(cls, file_path):
+        """
+        This function converts a hdf5 file to a Radial1DModel Object.
+
+        Parameters
+        ----------
+
+        file_path : `str`
+            path to Simulation generated hdf file
+
+        Returns
+        -------
+
+        model : `~Radial1DModel`
+        """
+
+        if file_path is None:
+            raise ValueError("File Path can`t be None")
+        if not os.path.exists(file_path):
+            raise IOError("Supplied HDF5 File %s does not exists" % file_path)
+
+        with h5py.File(file_path, 'r') as h5_file:
+            sim_dict = {}  # It will store HDF Objects in a dict form
+            for simulation in h5_file.keys():
+                for key in h5_file[simulation]:
+                    sim_dict[key] = {}
+                    if 'model' in key:
+                        cls.read_model_data(
+                            h5_file, simulation + '/model/', sim_dict[key], file_path)
+                    if 'plasma' in key:
+                        cls.read_plasma_data(
+                            h5_file, simulation + '/plasma/', sim_dict[key], file_path)
+
+        #Creates corresponding astropy.units.Quantity objects
+        abundance = sim_dict['plasma']['abundance']
+        time_explosion = sim_dict['plasma']['scalars']['time_explosion'] * u.s
+        t_inner = sim_dict['model']['scalars']['t_inner'] * u.K
+        t_radiative = np.array(sim_dict['plasma']['t_rad']) * u.K
+        v_boundary_inner = sim_dict['model']['v_inner'][0] * u.cm / u.s
+        v_boundary_outer = sim_dict['model']['v_outer'][
+            len(sim_dict['model']['v_outer']) - 1] * u.cm / u.s
+        dilution_factor = np.array(sim_dict['model']['w'])
+        velocity = np.append(sim_dict['model']['v_inner'],
+                             v_boundary_outer.value) * u.cm / u.s
+
+        # Presently homologous_density and luminosity_requested parameters are
+        # set to None
+        model = Radial1DModel(velocity, None, abundance, time_explosion,
+                              t_inner, None, t_radiative,
+                              dilution_factor, v_boundary_inner,
+                              v_boundary_outer)
+        # TODO : Extend it to plasma and montecarlo objects and return Simulation object
+
+        return model
+
+    @classmethod
+    def read_model_data(cls, h5_file, path, sim_dict, file_path):
```

Comment:
For the renaming part ,` read_model_data` , just stores attributes in a nested dictionary , and it does not return anything. Renaming it to `from_hdf `under `Radial1DModel` class will be misleading  , as it can mean, it returns `Radial1DModel `object .
So , I think , it will be better if I move it to `Radial1DModel `, without renaming.

I cannot return `Radial1DModel` ,just by reading model attributes in HDF file .
Because , some attributes, which are required for generating this object , also resides in `plasma` and `runner `path.